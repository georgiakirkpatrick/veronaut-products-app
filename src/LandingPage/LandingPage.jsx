import React, { useState } from 'react'
import CatCarousel from '../CatCarousel/CatCarousel'
import EmailSignUp from '../EmailSignUp/EmailSignUp'
import PrincipleList from '../PrincipleList/PrincipleList'
import ProductListPage from '../ProductListPage/ProductListPage'
import './LandingPage.css'

const LandingPage = ({
  catArray = [
    {
      id: 0,
      text: '',
      value: 0,
      class: '',
      featureImage: ''
    }
  ],
  principles = []
}) => {
  
  const [newProdArray, setNewProdArray] = useState([])

  // const principles = [
  //     {
  //         id: 1,
  //         title: 'Sustainability is subjective', 
  //         description: 'We’re not going to tell you which product to purchase or which brand to boycott.  What makes a garment’s supply chain “sustainable” is not only a complex issue, it also depends on who you ask.  We want to help you make informed decisions by making it as easy as possible to find the facts.',
  //         symbol: 'leaf'
  //     },
  //     {
  //         id: 2,
  //         title: 'Product research, but make it shopping',
  //         description: 'Finding the right clothes to fit your style, budget, and body can be tricky.  Add in your ethical-production standards and it can be downright frustrating to find products that check all the boxes.  That’s why we present product information in a way that feels like online shopping.  We think understanding the socioeconomic and environmental impacts of your purchase should be part of your shopping experience rather than an extra step.',
  //         symbol: 'tshirt'
  //     }
  // ]

  const makeCategorySlug = cat => (
    cat.text
      .normalize("NFD")
      .replace(/[\u0300-\u036f]/g, "")
      .replace(/\s+/g, '-')
      .toLowerCase()
  )

  const catImages = catArray.map(cat => (
    {
      id: cat.id,
      category: cat.text,
      class: cat.category_class,
      slug: makeCategorySlug(cat),
      url: cat.featureImage
    }
  ))

  return (
    <section className='LandingPage'>
      <div className='LandingPage__hero'>
        <section className='LandingPage__hero-title'>
          <header>
            <h1>
              <span className='fashion'>TRUTH </span>
              <br></br>
              <span className='fashion'>IS IN FASHION</span>
            </h1>
          </header>

          <p className='subhero'>
            <span className='subero1'>We're here to help you</span>
            <br></br>
            <span className='subero2'>find it</span>
          </p>
        </section>
      </div>
            
      <div className='LandingPage__body'>
        <section className='LandingPage__categories'>
          <header>
            <h2>Categories</h2>
          </header>
          
          <CatCarousel id='category-carousel' images={catImages}/>
        </section>

        <PrincipleList
          principles={principles}
        />

        <section className='LandingPage__new-products'>
          <ProductListPage
            catArray={catArray}
            prodArray={newProdArray}
            setProdArray={setNewProdArray}
          />
        </section>
      </div>

      <EmailSignUp />
    </section>
  )
}

export default LandingPage