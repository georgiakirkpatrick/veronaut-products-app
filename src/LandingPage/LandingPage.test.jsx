import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import { BrowserRouter as Router } from 'react-router-dom'
import LandingPage from './LandingPage'

describe ('LandingPage', () => {
  it ('renders without crashing', () => {
    render(<LandingPage />)
    screen.debug()
  })

  it('renders the UI as expected', () => {
    const tree = render(<LandingPage />)
    expect(tree).toMatchSnapshot()    
  })
})