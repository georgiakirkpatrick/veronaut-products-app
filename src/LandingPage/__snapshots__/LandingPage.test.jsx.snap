// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LandingPage > renders the UI as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="LandingPage"
      >
        <div
          class="LandingPage__hero"
        >
          <section
            class="LandingPage__hero-title"
          >
            <header>
              <h1>
                <span
                  class="fashion"
                >
                  TRUTH 
                </span>
                <br />
                <span
                  class="fashion"
                >
                  IS IN FASHION
                </span>
              </h1>
            </header>
            <p
              class="subhero"
            >
              <span
                class="subero1"
              >
                We're here to help you
              </span>
              <br />
              <span
                class="subero2"
              >
                find it
              </span>
            </p>
          </section>
        </div>
        <div
          class="LandingPage__body"
        >
          <section
            class="LandingPage__categories"
          >
            <header>
              <h2>
                Categories
              </h2>
            </header>
            <div
              class="CatCarousel"
              id="category-carousel"
            >
              <ul
                class="CatCarousel__list"
              >
                <li>
                  <div>
                    <img
                      alt=""
                      src=""
                    />
                  </div>
                  <p />
                </li>
              </ul>
            </div>
          </section>
          <section
            class="PrincipleList"
          >
            <header>
              <h2>
                Transparency is key
              </h2>
              <p>
                Veronaut helps you understand product descriptions and points out the information brands choose to omit.
              </p>
            </header>
            <ul />
          </section>
          <section
            class="LandingPage__new-products"
          >
            <div>
               Loading... 
            </div>
          </section>
        </div>
        <section
          class="EmailSignUp"
        >
          <header>
            <h2>
              Save your favorite products
            </h2>
          </header>
          <p>
            Create an account to save your favorite products.
          </p>
          <form>
            <fieldset>
              <label
                for="email-signup"
              >
                Email 
              </label>
              <input
                id="email-signup"
                name="email-signup"
                type="text"
              />
            </fieldset>
            <button
              class="FormButton"
              type="button"
            >
              Create account
            </button>
          </form>
        </section>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="LandingPage"
    >
      <div
        class="LandingPage__hero"
      >
        <section
          class="LandingPage__hero-title"
        >
          <header>
            <h1>
              <span
                class="fashion"
              >
                TRUTH 
              </span>
              <br />
              <span
                class="fashion"
              >
                IS IN FASHION
              </span>
            </h1>
          </header>
          <p
            class="subhero"
          >
            <span
              class="subero1"
            >
              We're here to help you
            </span>
            <br />
            <span
              class="subero2"
            >
              find it
            </span>
          </p>
        </section>
      </div>
      <div
        class="LandingPage__body"
      >
        <section
          class="LandingPage__categories"
        >
          <header>
            <h2>
              Categories
            </h2>
          </header>
          <div
            class="CatCarousel"
            id="category-carousel"
          >
            <ul
              class="CatCarousel__list"
            >
              <li>
                <div>
                  <img
                    alt=""
                    src=""
                  />
                </div>
                <p />
              </li>
            </ul>
          </div>
        </section>
        <section
          class="PrincipleList"
        >
          <header>
            <h2>
              Transparency is key
            </h2>
            <p>
              Veronaut helps you understand product descriptions and points out the information brands choose to omit.
            </p>
          </header>
          <ul />
        </section>
        <section
          class="LandingPage__new-products"
        >
          <div>
             Loading... 
          </div>
        </section>
      </div>
      <section
        class="EmailSignUp"
      >
        <header>
          <h2>
            Save your favorite products
          </h2>
        </header>
        <p>
          Create an account to save your favorite products.
        </p>
        <form>
          <fieldset>
            <label
              for="email-signup"
            >
              Email 
            </label>
            <input
              id="email-signup"
              name="email-signup"
              type="text"
            />
          </fieldset>
          <button
            class="FormButton"
            type="button"
          >
            Create account
          </button>
        </form>
      </section>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
